iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -234,22 +240,23 @@ private static void checkResiduals(CombinedScanTask task) {
     private CloseableIterator<T> currentIterator;
     private FileIO io;
     private EncryptionManager encryptionManager;
+    private Table table;
```

Comment:
Do we need the whole table here? Or we just need the partition objects and schema and...

Either we remove the table, and set the specific values in `initialize`, or keep the table and remove the ones which are easily accessible, and do not need calculation.

Reply:
We need the whole table object for this call:
```
public static NestedField metadataColumn(Table table, String name)
```
(which is called inside IcebergAcidUtil#createFileReadSchemaForDelete)
-> this gives us the _partition metadata column during file read

Reply:
I'll remove those fields which are easily derivable from table, such as io and encryption